### PR TITLE
Fix/condo/doma 3536/deleting meter readings by property id script

### PR DESCRIPTION
**Pull request by 2000Ilya**
Deleting meters and meter readings by IDs of properties from db script.

### CODE REVIEW

**Review by VKislov — APPROVED** [review]

File: apps/condo/bin/delete-meter-readings-by-property-ids.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        const { distDir, keystone, apps } = require(resolved)
+        const graphqlIndex = apps.findIndex(app => app instanceof GraphQLApp)
+        await keystone.prepare({ apps: [apps[graphqlIndex]], distDir, dev: true })
+        await keystone.connect()
+        this.context = await keystone.createContext({ skipAccessControl: true })
+    }
+
+    async findMeterReadings () {
+        this.meterReadings = await find('MeterReading', {
+            meter: { property: { id_in: ['3377a303-b098-426a-b559-e0acdf57071e', '19e7b8f0-d204-4f1e-80df-6069f01b15b4'] } },
```
Comment (SavelevMatthew) [inline apps/condo/bin/delete-meter-readings-by-property-ids.js:23]: May be pass property id as script args? And throw error is not passed?

Comment (2000Ilya) [reply apps/condo/bin/delete-meter-readings-by-property-ids.js]: propertyIds variable and throwing of exception added

File: apps/condo/bin/delete-meter-readings-by-property-ids.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    }
+    const fixer = new FixMeterReadingsClients(propertyIds)
+    console.info('[INFO] Connecting to database...')
+    await fixer.connect()
+    console.info('[INFO] Finding broken meter readings and meters...')
+    await fixer.findMeterReadings()
+    console.info(`[INFO] Following meter readings will be deleted: [${fixer.meterReadings.map(reading => `'${reading.id}'`).join(', ')}]`)
+    await fixer.findMeters()
+    console.info(`[INFO] Following meters will be deleted: [${fixer.meters.map(reading => `'${reading.id}'`).join(', ')}]`)
+    await fixer.fixBrokenMeterReadings()
```
Comment (akarjakin) [inline apps/condo/bin/delete-meter-readings-by-property-ids.js:73]: this method is missing

File: apps/condo/bin/delete-meter-readings-by-property-ids.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    console.info('[INFO] Broken meter readings are deleted...')
+    await fixer.fixBrokenMeters()
```
Comment (akarjakin) [inline apps/condo/bin/delete-meter-readings-by-property-ids.js:75]: this method is missing

**Review by akarjakin — CHANGES_REQUESTED** [review]

File: apps/condo/bin/delete-meter-readings-by-property-ids.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
+const path = require('path')
```
Comment (pahaz) [inline apps/condo/bin/delete-meter-readings-by-property-ids.js:1]: you also need to add ./bin/README.md example

File: apps/condo/bin/delete-meter-readings-by-property-ids.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        this.meters = await find('Meter', {
+            property: { id_in: this.propertyIds },
+        })
+    }
+
+    async deleteBrokenMeterReadings () {
+        if (!get(this.meterReadings, 'length')) return
+
+        for (const meterReading of this.meterReadings) {
+            await MeterReading.update(this.context, meterReading.id, {
```
Comment (AntonAL) [inline apps/condo/bin/delete-meter-readings-by-property-ids.js:40]: We have `softDelete` method in `generate.server.utils.js` module.

File: apps/condo/bin/delete-meter-readings-by-property-ids.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+                deletedAt: 'true',
+                dv: 1,
+                sender: { dv: 1, fingerprint: 'deleteIncorrectMeterReadingsScript' },
+            })
+        }
+    }
+
+    async deleteBrokenMeters () {
+        if (!get(this.meters, 'length')) return
```
[the remark this answers is not in the document]
Comment (2000Ilya) [reply apps/condo/bin/delete-meter-readings-by-property-ids.js]: added fix

File: apps/condo/bin/delete-meter-readings-by-property-ids.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+        })
+    }
+
+    async findMeters () {
+        this.meters = await find('Meter', {
+            property: { id_in: this.propertyIds },
+        })
+    }
+
+    async deleteBrokenMeterReadings () {
```
[the remark this answers is not in the document]
Comment (2000Ilya) [reply apps/condo/bin/delete-meter-readings-by-property-ids.js]: changed name of method

File: apps/condo/bin/delete-meter-readings-by-property-ids.js (hunk 4 above)
Comment (2000Ilya) [reply apps/condo/bin/delete-meter-readings-by-property-ids.js:1]: README.md example added

File: apps/condo/bin/delete-meter-readings-by-property-ids.js (hunk 5 above)
Comment (2000Ilya) [reply apps/condo/bin/delete-meter-readings-by-property-ids.js]: added softDelete method call

**Review by akarjakin — APPROVED** [review]

**Pull request merged**